src/rgw/rgw_coroutine.cc

```diff
@@ -570,6 +570,9 @@ void RGWCoroutinesManager::handle_unblocked_stack(set<RGWCoroutinesStack *>& con
   if (stack->is_io_blocked()) {
     --(*blocked_count);
     stack->set_io_blocked(false);
+    if (stack->is_interval_waiting()) {
+      --(*interval_wait_count);
+    }
```

Comment:
nesting this inside of the `is_io_blocked()` block assumes that interval_waits will always be io_blocked. this appears to be the case for `RGWCoroutinesStack::wait()`, but maybe this logic shouldn't treat them independently?
```diff
   if (stack->is_io_blocked()) {
     --(*blocked_count);
     stack->set_io_blocked(false);
   }
+  if (stack->is_interval_waiting()) {
+    --(*interval_wait_count);
+    stack->set_interval_wait(false);
+  }
-  stack->set_interval_wait(false);
```

Reply:
Maybe, at present , the code shows interval_wait is one of scene of io_block

Reply:
RGWCoroutinesManager::run.  
    if (stack->is_io_blocked())
    {
      ldout(cct, 20) << __func__ << ":"
                     << " stack=" << (void *)stack << " is io blocked" << dendl;
      if (stack->is_interval_waiting())
      {
        interval_wait_count++;
      }
      blocked_count++;
    }

may this logic should treat them independently ?

Reply:
@cbodley  there seems to be one case where we want to call handle_unblocked_stack() only on blocked operations without the interval_wait. Like in https://github.com/ceph/ceph/blob/8669e4de2a65dd0651e86421665143eaec061f5c/src/rgw/rgw_coroutine.cc#L717
In this case, we should probably not decrement interval_wait_count?

Reply:
@yehudasa if any suggestion?

Reply:
> @cbodley there seems to be one case where we want to call handle_unblocked_stack() only on blocked operations without the interval_wait. Like in
> 
> https://github.com/ceph/ceph/blob/8669e4de2a65dd0651e86421665143eaec061f5c/src/rgw/rgw_coroutine.cc#L717
> 
> 
> In this case, we should probably not decrement interval_wait_count?

in 

void RGWCoroutinesManager::handle_unblocked_stack(set<RGWCoroutinesStack *> &context_stacks, list<RGWCoroutinesStack *> &scheduled_stacks,
                                                  RGWCompletionManager::io_completion &io, int *blocked_count)
{
  ceph_assert(ceph_mutex_is_wlocked(lock));
  RGWCoroutinesStack *stack = static_cast<RGWCoroutinesStack *>(io.user_info);
  if (context_stacks.find(stack) == context_stacks.end())
  {
    return;
  }
  if (!stack->try_io_unblock(io.io_id))
  {
    return;
  }
  if (stack->is_io_blocked())
  {
    --(*blocked_count);
    stack->set_io_blocked(false);
  }
  stack->set_interval_wait(false);
  if (!stack->is_done())
  {
    if (!stack->is_scheduled)
    {
      scheduled_stacks.push_back(stack);
      stack->set_is_scheduled(true);
    }
  }
  else
  {
    context_stacks.erase(stack);
    stack->put();
  }
}

call  stack->set_interval_wait(false) to change stack state to  not_interval_wait, should dec interval_wait_count?